agreementbot/api.go

```diff
@@ -442,6 +446,31 @@ func (a *API) agreement(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// Obscure the secret details for all secrets in the given proposal
+func obscureSecretDetails(p *persistence.Agreement) error {
+	prop := p.Proposal
+
+	var tsandcs map[string]interface{}
+	if err := json.Unmarshal([]byte(prop), &tsandcs); err != nil {
```

Comment:
There is a function for demarshalling proposals already: abstractprotocol.DemarshalProposal(proposal), and there's a function for demarshalling the TsAndCs: policy.DemarshalPolicy(proposal.TsAndCs()). I would put the code to obscure the secrets in ./policy/policy_file.go. The Policy type represents the internal policy objects that Patterns and Deployment Policies get converted to.